images/base/files/usr/local/bin/entrypoint

```diff
@@ -101,22 +101,30 @@ overlayfs_preferrable() {
 
 configure_containerd() {
   local snapshotter=${KIND_EXPERIMENTAL_CONTAINERD_SNAPSHOTTER:-}
-  if [[ -n "$userns" ]]; then
-    # userns (rootless) configs
 
-    # Adjust oomScoreAdj
+  # handle userns (rootless)
+  if [[ -n "$userns" ]]; then
+    # enable restrict_oom_score_adj
     sed -i 's/restrict_oom_score_adj = false/restrict_oom_score_adj = true/' /etc/containerd/config.toml
-
     # Use fuse-overlayfs if overlayfs is not preferrable: https://github.com/kubernetes-sigs/kind/issues/2275
     if [[ -z "$snapshotter" ]] && ! overlayfs_preferrable; then
       snapshotter="fuse-overlayfs"
     fi
-  else
+  fi
+
+  # if we have not already overridden the snapshotter, attempt to auto select
+  if [[ -z "$snapshotter" ]]; then
     # we need to switch to the 'native' snapshotter on zfs
-    if [[ -z "$snapshotter" ]] && [[ "$(stat -f -c %T /kind)" == 'zfs' ]]; then
+    container_filesystem="$(stat -f -c %T /kind)"
+    if [[ "$container_filesystem" == 'zfs' ]]; then
       snapshotter="native"
+    # fuse likely implies overlayfs-fuse, we should switch to overlayfs-fuse (or native)
+    elif [[ "$container_filesystem" == 'fuseblk' ]]; then
```

Comment:
this is the functional difference, the rest is rearranging a little.

we will select fuse-overlayfs if we detect we're in fuseblk and we haven't otherwise already selected a snapshotter due to userns or KIND_EXPERIMENTAL_CONTAINERD_SNAPSHOTTER

Reply:
I don't _think_ there's any supported snapshotter environment where we'd be in fuse within the container filesystem but not using fuse-overlayfs.

Technically the user could configure a mount over `/kind` but that's already an unsafe choice, we should perhaps but a README in there to let people know that this is an internal directory.